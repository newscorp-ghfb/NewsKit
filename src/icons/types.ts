import {ComponentType} from 'react';
import {MQ} from '../utils/style/types';
import {Theme} from '../theme';
import {LogicalProps} from '../utils/logical-properties';

export interface SvgProps extends React.SVGAttributes<SVGElement> {
  title?: string;
}

export interface NewsKitIconProps extends SvgProps {
  overrides?: {
    stylePreset?: MQ<string>;
    size?: MQ<string>;
  } & LogicalProps;
}

export type NewsKitIcon = ComponentType<
  NewsKitIconProps & {theme?: Theme | undefined}
>;
