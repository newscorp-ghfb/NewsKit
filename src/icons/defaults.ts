export default {
  icons: {
    stylePreset: 'iconDefault',
  },
};
