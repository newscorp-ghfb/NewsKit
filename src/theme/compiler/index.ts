/* eslint-disable @typescript-eslint/no-explicit-any */
import {Theme, UncompiledTheme} from '../types';
import {ThemeCompilerOptions} from './types';
import {generateInkStylePresets} from './ink-style-preset-generator';
import {recurseUnknown} from '../../utils/recurse-unknown';

export const compileTheme = (
  theme: UncompiledTheme | Theme,
  options: ThemeCompilerOptions = {},
): Theme => {
  if (!theme || theme.compiled === true) {
    return theme;
  }

  // eslint-disable-next-line no-console
  const errorLogger = options.errorLogger || console.error.bind(console);

  const uncompiledTheme = {
    ...theme,
    stylePresets: {
      ...theme.stylePresets,
      ...generateInkStylePresets(theme),
    },
  };

  const {icons = {}, ...filteredTheme} = uncompiledTheme;
  return {
    ...(recurseUnknown(uncompiledTheme, filteredTheme, errorLogger) as any),
    icons,
    compiled: true,
  };
};
