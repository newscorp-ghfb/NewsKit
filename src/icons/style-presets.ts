import {StylePreset} from '../theme/types';

export default {
  iconTEST: {
    base: {
      iconColor: '{{colors.red050}}',
    },
    hover: {
      iconColor: '{{colors.green050}}',
    },
  },

  iconPositive: {
    base: {
      iconColor: '{{colors.inkPositive}}',
    },
  },
  iconNegative: {
    base: {
      iconColor: '{{colors.inkNegative}}',
    },
  },
  iconSocialTwitter: {
    base: {
      iconColor: '{{colors.socialTwitter}}',
    },
  },
  iconSocialGitHub: {
    base: {
      iconColor: '{{colors.socialGithub}}',
    },
  },
  iconSocialFacebook: {
    base: {
      iconColor: '{{colors.socialFacebook}}',
    },
  },
  iconSocialInstagram: {
    base: {
      iconColor: '{{colors.socialInstagram}}',
    },
  },
  iconSocialYoutube: {
    base: {
      iconColor: '{{colors.socialYoutube}}',
    },
  },
  iconSocialWhatsApp: {
    base: {
      iconColor: '{{colors.socialWhatsapp}}',
    },
  },
  iconSocialReddit: {
    base: {
      iconColor: '{{colors.socialReddit}}',
    },
  },

  iconDefault: {
    base: {
      iconColor: '{{colors.inkBase}}',
    },
  },
  indeterminateProgressIndicatorPrimary: {
    base: {
      iconColor: '{{colors.inkBrand010}}',
    },
  },
  indeterminateProgressIndicatorSecondary: {
    base: {
      iconColor: '{{colors.inkBase}}',
    },
  },
  indeterminateProgressIndicatorNegative: {
    base: {
      iconColor: '{{colors.inkNegative}}',
    },
  },
  indeterminateProgressIndicatorPositive: {
    base: {
      iconColor: '{{colors.inkPositive}}',
    },
  },
  indeterminateProgressIndicatorInverse: {
    base: {
      iconColor: '{{colors.inkInverse}}',
    },
  },
} as Record<string, StylePreset>;
