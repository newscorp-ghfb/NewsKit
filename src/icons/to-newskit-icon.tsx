import React from 'react';
import {EmotionIconProps} from '@emotion-icons/emotion-icon';
import {withTheme} from '../theme';
import {NewsKitIconProps, NewsKitIcon, SvgProps} from './types';
import {getSizingCssFromTheme, getStylePreset, styled} from '../utils/style';
import defaults from './defaults';
import stylePresets from './style-presets';
import {withOwnTheme} from '../utils/with-own-theme';
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const renderIconStylePreset = (overridesOnly: boolean) => (props: any) => {
  if (!overridesOnly || (props.overrides && props.overrides.stylePreset)) {
    return getStylePreset('icons', '', {
      isSvg: true,
    })(props);
  }
  return {};
};

export const toNewsKitIcon = (
  PassedIcon:
    | React.ComponentType<EmotionIconProps>
    | React.ComponentType<SvgProps>,
): NewsKitIcon =>
  withOwnTheme(
    withTheme<NewsKitIconProps>(
      React.memo(props => {
        const emotionIconName = PassedIcon.displayName;
        const toStyledIcon = (
          Icon: React.ComponentType<EmotionIconProps>,
        ) => styled(Icon)`
          // If not overridden, render SP CSS here, this allows parent SP to override Icon default.
          ${renderIconStylePreset(false)}

          vertical-align: unset;
          display: inline-block;
          // https://css-tricks.com/the-sass-ampersand/#doubling-up-specificity
          && {
            //we don't want the icon to have a default size hence using non defaulted functions
            ${props.overrides?.size &&
            getSizingCssFromTheme('width', props.overrides.size)}
            ${props.overrides?.size &&
            getSizingCssFromTheme('height', props.overrides.size)}
        // If overridden, render SP CSS here instead - this ensures we override fill color from parent SP.
        ${renderIconStylePreset(true)}
          }
        `;
        const Icon =
          props.theme.icons[`${emotionIconName}`] ||
          toStyledIcon(PassedIcon as React.ComponentType<EmotionIconProps>);
        return <Icon title={props.title} {...props} />;
      }),
    ),
  )({defaults, stylePresets});
