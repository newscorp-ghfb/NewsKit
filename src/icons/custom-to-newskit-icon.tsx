import {withDefaultProps} from '../utils/with-default-props';
import {toNewsKitIcon} from './to-newskit-icon';
import {NewsKitIconProps, SvgProps} from './types';

export const customToNewsKitIcon = (
  displayName: string,
  CustomIcon: React.ComponentType<SvgProps>,
  overrides?: NewsKitIconProps['overrides'],
) => {
  // eslint-disable-next-line no-param-reassign
  CustomIcon.displayName = displayName;

  if (!overrides) {
    return toNewsKitIcon(CustomIcon);
  }

  return withDefaultProps(toNewsKitIcon(CustomIcon), {
    overrides,
  });
};
