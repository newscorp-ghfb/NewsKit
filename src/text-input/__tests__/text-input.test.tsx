import React, {createRef} from 'react';
import {fireEvent} from '@testing-library/react';
import {TextInput, TextInputProps} from '..';
import {
  renderToFragmentWithTheme,
  renderWithTheme,
} from '../../test/test-utils';
import {createTheme} from '../../theme';
import {IconFilledSearch} from '../../icons';

const renderTextInput = (props: TextInputProps) => (
  <TextInput
    assistiveText="Assistive Text"
    data-testid="text-input-email"
    {...props}
  />
);

describe('TextInput', () => {
  test('renders correctly with label and assistive text', () => {
    const props: TextInputProps = {
      label: 'label',
    };
    const fragment = renderToFragmentWithTheme(renderTextInput, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders correctly without asistive text', () => {
    const props: TextInputProps = {
      label: 'label',
      spellCheck: true,
    };
    const fragment = renderToFragmentWithTheme(TextInput, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders correctly with label hidden', () => {
    const props: TextInputProps = {
      label: 'label',
      hideLabel: true,
    };
    const fragment = renderToFragmentWithTheme(renderTextInput, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders small text input', () => {
    const props: TextInputProps = {
      size: 'small',
      label: 'label',
    };
    const fragment = renderToFragmentWithTheme(renderTextInput, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders large text input', () => {
    const props: TextInputProps = {
      size: 'large',
      label: 'label',
    };
    const fragment = renderToFragmentWithTheme(renderTextInput, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders disabled text input', () => {
    const props: TextInputProps = {
      disabled: true,
      label: 'label',
    };
    const fragment = renderToFragmentWithTheme(renderTextInput, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders readonly text input', () => {
    const props: TextInputProps = {
      readOnly: true,
      label: 'label',
    };
    const fragment = renderToFragmentWithTheme(renderTextInput, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders with different minHeight', () => {
    const props: TextInputProps = {
      label: 'label',
      overrides: {
        input: {
          minHeight: 'sizing090',
        },
      },
    };
    const fragment = renderToFragmentWithTheme(renderTextInput, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders with fixed width', () => {
    const props: TextInputProps = {
      label: 'label',
      overrides: {
        width: 'sizing120',
      },
    };
    const fragment = renderToFragmentWithTheme(renderTextInput, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders with full width', () => {
    const props: TextInputProps = {
      label: 'label',
      overrides: {
        width: '100%',
      },
    };
    const fragment = renderToFragmentWithTheme(renderTextInput, props);
    expect(fragment).toMatchSnapshot();
  });

  test('should render custom aria-label', () => {
    const props: TextInputProps = {
      label: 'label',
      ariaLabel: 'my custom aria label',
    };

    const fragment = renderToFragmentWithTheme(renderTextInput, props);
    expect(fragment).toMatchSnapshot();
  });

  test('should render custom placeholder', () => {
    const props: TextInputProps = {
      label: 'label',
      placeholder: 'This is some text',
    };

    const fragment = renderToFragmentWithTheme(renderTextInput, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders text input with overrides', () => {
    const myCustomTheme = createTheme({
      name: 'my-custom-input-theme',
      overrides: {
        stylePresets: {
          textInputLabelCustom: {
            base: {
              color: '#f000dc',
            },
          },
          textInputContainerCustom: {
            base: {
              borderColor: '#D20600',
            },
          },
          assistiveTextCustom: {
            base: {
              color: '#f000dc',
            },
          },
        },
      },
    });
    const props: TextInputProps = {
      label: 'label',
      overrides: {
        label: {
          stylePreset: 'textInputLabelCustom',
          typographyPreset: 'utilityLabel010',
          spaceStack: 'space050',
          spaceInline: 'space020',
        },
        input: {
          stylePreset: 'textInputContainerCustom',
          spaceInset: 'spaceInset020',
          minHeight: 'sizing090',
          typographyPreset: 'utilityBody030',
          spaceStack: 'space040',
        },
        assistiveText: {
          stylePreset: 'assistiveTextCustom',
          typographyPreset: 'utilityLabel030',
          minHeight: 'sizing050',
        },
      },
    };
    const fragment = renderToFragmentWithTheme(
      renderTextInput,
      props,
      myCustomTheme,
    ) as any;

    expect(fragment).toMatchSnapshot();
  });

  it('focus can be triggered with ref', () => {
    const inputRef = createRef<HTMLInputElement>();

    const props = {
      label: 'label',
      ref: inputRef,
    };

    renderWithTheme(TextInput, props);

    if (inputRef && inputRef.current) {
      inputRef.current.focus();
    }
    expect(inputRef.current).toHaveFocus();
  });

  it('calls onBlur passed from the props', () => {
    const onBlur = jest.fn();
    const props: TextInputProps = {
      label: 'label',
      placeholder: 'This is some text',
      onBlur,
    };

    const {getByTestId} = renderWithTheme(renderTextInput, props);
    fireEvent.blur(getByTestId('text-input-email'));

    expect(onBlur).toHaveBeenCalled();
  });

  it('calls onChange passed from the props', () => {
    const onChange = jest.fn();
    const props: TextInputProps = {
      label: 'label',
      placeholder: 'This is some text',
      onChange,
    };

    const {getByTestId} = renderWithTheme(renderTextInput, props);
    fireEvent.change(getByTestId('text-input-email'), {
      target: {value: 'test'},
    });

    expect(onChange).toHaveBeenCalled();
  });

  test('renders text input with leading icon', () => {
    const props: TextInputProps = {
      label: 'label',
      icon: (
        <IconFilledSearch
          overrides={{
            size: 'iconSize020',
            stylePreset: 'inkSubtle',
          }}
        />
      ),
    };
    const fragment = renderToFragmentWithTheme(renderTextInput, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders text input with overrides for leading and validation icon', () => {
    const props: TextInputProps = {
      label: 'label',
      icon: (
        <IconFilledSearch
          overrides={{
            size: 'iconSize020',
            stylePreset: 'inkSubtle',
          }}
        />
      ),
      overrides: {
        input: {
          leadingIcon: {
            iconOffset: 'space040',
            spaceInset: 'space080',
          },
          validationIcon: {
            iconOffset: 'space040',
            spaceInset: 'space080',
            iconSize: 'iconSize030',
          },
        },
      },
    };
    const fragment = renderToFragmentWithTheme(renderTextInput, props);
    expect(fragment).toMatchSnapshot();
  });
});
