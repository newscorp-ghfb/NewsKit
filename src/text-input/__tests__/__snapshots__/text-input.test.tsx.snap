// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly with label and assistive text 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: block;
}

.emotion-2 {
  position: relative;
}

.emotion-3 {
  box-sizing: border-box;
  width: 100%;
  cursor: default;
}

.emotion-5 {
  margin: 0;
}

<div
    class="emotion-0"
  >
    <label
      class="emotion-1"
      for="mock-nk-1"
    >
      label
    </label>
    <div
      class="emotion-2"
    >
      <input
        aria-describedby="mock-nk-1-assistive-text"
        aria-invalid="false"
        class="emotion-3"
        data-testid="text-input-email"
        id="mock-nk-1"
        spellcheck="false"
        type="text"
      />
    </div>
    <div
      class="emotion-4"
    >
      <p
        class="emotion-5"
        id="mock-nk-1-assistive-text"
      >
        Assistive Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput renders correctly with label hidden 1`] = `
<DocumentFragment>
  .emotion-1 {
  position: relative;
}

.emotion-2 {
  box-sizing: border-box;
  width: 100%;
  cursor: default;
}

.emotion-4 {
  margin: 0;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <input
        aria-describedby="mock-nk-1-assistive-text"
        aria-invalid="false"
        aria-label="label"
        class="emotion-2"
        data-testid="text-input-email"
        id="mock-nk-1"
        spellcheck="false"
        type="text"
      />
    </div>
    <div
      class="emotion-3"
    >
      <p
        class="emotion-4"
        id="mock-nk-1-assistive-text"
      >
        Assistive Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput renders correctly without asistive text 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: block;
}

.emotion-2 {
  position: relative;
}

.emotion-3 {
  box-sizing: border-box;
  width: 100%;
  cursor: default;
}

<div
    class="emotion-0"
  >
    <label
      class="emotion-1"
      for="mock-nk-1"
    >
      label
    </label>
    <div
      class="emotion-2"
    >
      <input
        aria-invalid="false"
        class="emotion-3"
        id="mock-nk-1"
        spellcheck="true"
        type="text"
      />
    </div>
    <div
      class="emotion-4"
    />
  </div>
</DocumentFragment>
`;

exports[`TextInput renders disabled text input 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: block;
}

.emotion-2 {
  position: relative;
}

.emotion-3 {
  box-sizing: border-box;
  width: 100%;
  cursor: not-allowed;
}

.emotion-5 {
  margin: 0;
}

<div
    class="emotion-0"
  >
    <label
      class="emotion-1"
      disabled=""
      for="mock-nk-1"
    >
      label
    </label>
    <div
      class="emotion-2"
    >
      <input
        aria-describedby="mock-nk-1-assistive-text"
        aria-invalid="false"
        class="emotion-3"
        data-testid="text-input-email"
        disabled=""
        id="mock-nk-1"
        spellcheck="false"
        type="text"
      />
    </div>
    <div
      class="emotion-4"
    >
      <p
        class="emotion-5"
        disabled=""
        id="mock-nk-1-assistive-text"
      >
        Assistive Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput renders large text input 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: block;
}

.emotion-2 {
  position: relative;
}

.emotion-3 {
  box-sizing: border-box;
  width: 100%;
  cursor: default;
}

.emotion-5 {
  margin: 0;
}

<div
    class="emotion-0"
  >
    <label
      class="emotion-1"
      for="mock-nk-1"
    >
      label
    </label>
    <div
      class="emotion-2"
    >
      <input
        aria-describedby="mock-nk-1-assistive-text"
        aria-invalid="false"
        class="emotion-3"
        data-testid="text-input-email"
        id="mock-nk-1"
        spellcheck="false"
        type="text"
      />
    </div>
    <div
      class="emotion-4"
    >
      <p
        class="emotion-5"
        id="mock-nk-1-assistive-text"
      >
        Assistive Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput renders readonly text input 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: block;
}

.emotion-2 {
  position: relative;
}

.emotion-3 {
  box-sizing: border-box;
  width: 100%;
  cursor: default;
}

.emotion-5 {
  margin: 0;
}

<div
    class="emotion-0"
  >
    <label
      class="emotion-1"
      for="mock-nk-1"
    >
      label
    </label>
    <div
      class="emotion-2"
    >
      <input
        aria-describedby="mock-nk-1-assistive-text"
        aria-invalid="false"
        class="emotion-3"
        data-testid="text-input-email"
        id="mock-nk-1"
        readonly=""
        spellcheck="false"
        type="text"
      />
    </div>
    <div
      class="emotion-4"
    >
      <p
        class="emotion-5"
        id="mock-nk-1-assistive-text"
      >
        Assistive Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput renders small text input 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: block;
}

.emotion-2 {
  position: relative;
}

.emotion-3 {
  box-sizing: border-box;
  width: 100%;
  cursor: default;
}

.emotion-5 {
  margin: 0;
}

<div
    class="emotion-0"
  >
    <label
      class="emotion-1"
      for="mock-nk-1"
    >
      label
    </label>
    <div
      class="emotion-2"
    >
      <input
        aria-describedby="mock-nk-1-assistive-text"
        aria-invalid="false"
        class="emotion-3"
        data-testid="text-input-email"
        id="mock-nk-1"
        spellcheck="false"
        type="text"
      />
    </div>
    <div
      class="emotion-4"
    >
      <p
        class="emotion-5"
        id="mock-nk-1-assistive-text"
      >
        Assistive Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput renders text input with leading icon 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: block;
}

.emotion-2 {
  position: relative;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(0, -50%);
  -moz-transform: translate(0, -50%);
  -ms-transform: translate(0, -50%);
  transform: translate(0, -50%);
  pointer-events: none;
}

.emotion-4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  color: #535353;
  fill: #535353;
  vertical-align: unset;
  display: inline-block;
}

.emotion-4.emotion-4 {
  width: 24px;
  height: 24px;
  color: #535353;
  fill: #535353;
}

.emotion-6 {
  box-sizing: border-box;
  width: 100%;
  cursor: default;
}

.emotion-8 {
  margin: 0;
}

<div
    class="emotion-0"
  >
    <label
      class="emotion-1"
      for="mock-nk-1"
    >
      label
    </label>
    <div
      class="emotion-2"
    >
      <span
        class="emotion-3"
      >
        <svg
          aria-hidden="true"
          class="emotion-4 emotion-5"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </span>
      <input
        aria-describedby="mock-nk-1-assistive-text"
        aria-invalid="false"
        class="emotion-6"
        data-testid="text-input-email"
        id="mock-nk-1"
        spellcheck="false"
        type="text"
      />
    </div>
    <div
      class="emotion-7"
    >
      <p
        class="emotion-8"
        id="mock-nk-1-assistive-text"
      >
        Assistive Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput renders text input with overrides 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: block;
  color: #f000dc;
  font-family: "Poppins",sans-serif;
  font-size: 12px;
  line-height: 18px;
  font-weight: 500;
  letter-spacing: -0.5px;
  padding: 0.5px 0px;
  margin-bottom: 24px;
  margin-right: 8px;
}

.emotion-1::before {
  content: '';
  margin-bottom: -0.403em;
  display: block;
}

.emotion-1::after {
  content: '';
  margin-top: -0.4em;
  display: block;
}

.emotion-2 {
  position: relative;
  margin-bottom: 16px;
}

.emotion-3 {
  font-family: "DM Sans",sans-serif;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;
  letter-spacing: -0.5px;
  padding: 0.5px 0px;
  box-sizing: border-box;
  width: 100%;
  cursor: default;
  border-color: #D20600;
  padding: 8px;
  min-height: 64px;
}

.emotion-3::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-3::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

.emotion-4 {
  min-height: 24px;
}

.emotion-5 {
  margin: 0;
  color: #f000dc;
  font-family: "Poppins",sans-serif;
  font-size: 16px;
  line-height: 24px;
  font-weight: 500;
  letter-spacing: -0.5px;
  padding: 0.5px 0px;
}

.emotion-5::before {
  content: '';
  margin-bottom: -0.403em;
  display: block;
}

.emotion-5::after {
  content: '';
  margin-top: -0.4em;
  display: block;
}

<div
    class="emotion-0"
  >
    <label
      class="emotion-1"
      for="mock-nk-1"
    >
      label
    </label>
    <div
      class="emotion-2"
    >
      <input
        aria-describedby="mock-nk-1-assistive-text"
        aria-invalid="false"
        class="emotion-3"
        data-testid="text-input-email"
        id="mock-nk-1"
        spellcheck="false"
        type="text"
      />
    </div>
    <div
      class="emotion-4"
    >
      <p
        class="emotion-5"
        id="mock-nk-1-assistive-text"
      >
        Assistive Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput renders text input with overrides for leading and validation icon 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: block;
}

.emotion-2 {
  position: relative;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(0, -50%);
  -moz-transform: translate(0, -50%);
  -ms-transform: translate(0, -50%);
  transform: translate(0, -50%);
  pointer-events: none;
}

.emotion-4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  color: #535353;
  fill: #535353;
  vertical-align: unset;
  display: inline-block;
}

.emotion-4.emotion-4 {
  width: 24px;
  height: 24px;
  color: #535353;
  fill: #535353;
}

.emotion-6 {
  box-sizing: border-box;
  width: 100%;
  cursor: default;
}

.emotion-6 {
  padding-left: 48px;
}

.emotion-8 {
  margin: 0;
}

<div
    class="emotion-0"
  >
    <label
      class="emotion-1"
      for="mock-nk-1"
    >
      label
    </label>
    <div
      class="emotion-2"
    >
      <span
        class="emotion-3"
      >
        <svg
          aria-hidden="true"
          class="emotion-4 emotion-5"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </span>
      <input
        aria-describedby="mock-nk-1-assistive-text"
        aria-invalid="false"
        class="emotion-6"
        data-testid="text-input-email"
        id="mock-nk-1"
        spellcheck="false"
        type="text"
      />
    </div>
    <div
      class="emotion-7"
    >
      <p
        class="emotion-8"
        id="mock-nk-1-assistive-text"
      >
        Assistive Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput renders with different minHeight 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: block;
}

.emotion-2 {
  position: relative;
}

.emotion-3 {
  box-sizing: border-box;
  width: 100%;
  cursor: default;
  min-height: 64px;
}

.emotion-5 {
  margin: 0;
}

<div
    class="emotion-0"
  >
    <label
      class="emotion-1"
      for="mock-nk-1"
    >
      label
    </label>
    <div
      class="emotion-2"
    >
      <input
        aria-describedby="mock-nk-1-assistive-text"
        aria-invalid="false"
        class="emotion-3"
        data-testid="text-input-email"
        id="mock-nk-1"
        spellcheck="false"
        type="text"
      />
    </div>
    <div
      class="emotion-4"
    >
      <p
        class="emotion-5"
        id="mock-nk-1-assistive-text"
      >
        Assistive Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput renders with fixed width 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 160px;
}

.emotion-1 {
  display: block;
}

.emotion-2 {
  position: relative;
}

.emotion-3 {
  box-sizing: border-box;
  width: 100%;
  cursor: default;
}

.emotion-5 {
  margin: 0;
}

<div
    class="emotion-0"
  >
    <label
      class="emotion-1"
      for="mock-nk-1"
    >
      label
    </label>
    <div
      class="emotion-2"
    >
      <input
        aria-describedby="mock-nk-1-assistive-text"
        aria-invalid="false"
        class="emotion-3"
        data-testid="text-input-email"
        id="mock-nk-1"
        spellcheck="false"
        type="text"
      />
    </div>
    <div
      class="emotion-4"
    >
      <p
        class="emotion-5"
        id="mock-nk-1-assistive-text"
      >
        Assistive Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput renders with full width 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
}

.emotion-1 {
  display: block;
}

.emotion-2 {
  position: relative;
}

.emotion-3 {
  box-sizing: border-box;
  width: 100%;
  cursor: default;
}

.emotion-5 {
  margin: 0;
}

<div
    class="emotion-0"
  >
    <label
      class="emotion-1"
      for="mock-nk-1"
    >
      label
    </label>
    <div
      class="emotion-2"
    >
      <input
        aria-describedby="mock-nk-1-assistive-text"
        aria-invalid="false"
        class="emotion-3"
        data-testid="text-input-email"
        id="mock-nk-1"
        spellcheck="false"
        type="text"
      />
    </div>
    <div
      class="emotion-4"
    >
      <p
        class="emotion-5"
        id="mock-nk-1-assistive-text"
      >
        Assistive Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput should render custom aria-label 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: block;
}

.emotion-2 {
  position: relative;
}

.emotion-3 {
  box-sizing: border-box;
  width: 100%;
  cursor: default;
}

.emotion-5 {
  margin: 0;
}

<div
    class="emotion-0"
  >
    <label
      class="emotion-1"
      for="mock-nk-1"
    >
      label
    </label>
    <div
      class="emotion-2"
    >
      <input
        aria-describedby="mock-nk-1-assistive-text"
        aria-invalid="false"
        aria-label="my custom aria label"
        class="emotion-3"
        data-testid="text-input-email"
        id="mock-nk-1"
        spellcheck="false"
        type="text"
      />
    </div>
    <div
      class="emotion-4"
    >
      <p
        class="emotion-5"
        id="mock-nk-1-assistive-text"
      >
        Assistive Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput should render custom placeholder 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: block;
}

.emotion-2 {
  position: relative;
}

.emotion-3 {
  box-sizing: border-box;
  width: 100%;
  cursor: default;
}

.emotion-5 {
  margin: 0;
}

<div
    class="emotion-0"
  >
    <label
      class="emotion-1"
      for="mock-nk-1"
    >
      label
    </label>
    <div
      class="emotion-2"
    >
      <input
        aria-describedby="mock-nk-1-assistive-text"
        aria-invalid="false"
        class="emotion-3"
        data-testid="text-input-email"
        id="mock-nk-1"
        placeholder="This is some text"
        spellcheck="false"
        type="text"
      />
    </div>
    <div
      class="emotion-4"
    >
      <p
        class="emotion-5"
        id="mock-nk-1-assistive-text"
      >
        Assistive Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;
